include/session/onionreq/builder.h

```diff
@@ -25,13 +25,21 @@ typedef struct onion_request_builder_object {
 ///
 /// Constructs an onion request builder and sets a pointer to it in `builder`.
 ///
-/// When done with the object the `builder` must be destroyed by either passing the pointer to
-/// onion_request_builder_free() or onion_request_builder_build().
+/// When done with the object the `builder` must be destroyed by passing the pointer to
+/// onion_request_builder_free().
```

Comment:
This comment was wrong (there was no builder_free function, the `_build()` function doesn't free anything).  And while _build *could* free, it seems a little weird to have it do so, because then you can't make one request builder and reuse it multiple times with different payloads.

The reason I comment though: if session-ios is calling this, it is worth double-checking that it isn't actually expecting `_build` to free the instance (and also adding a call to `_free` at the appropriate place).

Reply:
There was a point where I'd updated session-ios to use these but before it got merged we ended up plugging `libQuic` in so now all the onion request building happens internally on the C++ side of things - so probably good to fix but luckily it won't affect anything currently 🎉